iroh-net/src/hp/magicsock/conn.rs

```diff
@@ -202,16 +202,35 @@ pub struct Inner {
     /// None (or zero regions/nodes) means DERP is disabled.
     pub(super) derp_map: tokio::sync::RwLock<Option<DerpMap>>,
     /// Nearest DERP region ID; 0 means none/unknown.
-    my_derp: AtomicU16,
+    /// None means we are not connected or do not yet know the connection status.
+    // I'm sorry.
+    my_derp: tokio::sync::RwLock<Option<u16>>,
```

Comment:
the dreaded new lock.

Please feel free to reverse this decision.

Reply:
... I've already reversed the decision.

Reply:
what an emotional journey